templates/terraform/custom_delete/appversion_delete.go.erb

```diff
@@ -0,0 +1,77 @@
+
+if d.Get("noop_on_destroy") == true {
+	log.Printf("[DEBUG] Keeping the StandardAppVersion %q", d.Id())
+	return nil
+} else if d.Get("delete_service_on_destroy") == true {
+	config := meta.(*Config)
```

Comment:
We can reduce duplicated code and "unnecessary-else-after-return" here by removing the `else if` and `else`, since the code in `if` terminates the function. The first ten lines or so of the current `else if` and `else` blocks are duplicated. How about:

```
if noop {
   ...
   return nil
}
config := meta.(*Config)

project, err := getProject(d, config)
if err != nil {
    return err
}

lockName, err := replaceVars(d, config, "apps/{{project}}/services/{{service}}")
if err != nil {
    return err
}
mutexKV.Lock(lockName)
defer mutexKV.Unlock(lockName)

if delete_service_on_destroy {
    ...
} else {
   ...
}
```